vm/src/frame.rs

```diff
@@ -995,19 +1013,27 @@ impl Frame {
         Ok(None)
     }
 
-    fn load_name(&self, vm: &VirtualMachine, name: &str) -> FrameResult {
-        match self.scope.load_name(&vm, name) {
-            Some(value) => {
-                self.push_value(value);
-                Ok(None)
-            }
-            None => {
-                let name_error_type = vm.ctx.exceptions.name_error.clone();
-                let msg = format!("name '{}' is not defined", name);
-                let name_error = vm.new_exception(name_error_type, msg);
-                Err(name_error)
-            }
-        }
+    fn load_name(
+        &self,
+        vm: &VirtualMachine,
+        name: &str,
+        scope: &bytecode::NameScope,
+    ) -> FrameResult {
+        let value = match scope {
+            bytecode::NameScope::Global => self.scope.globals.get_item(name, vm)?,
```

Comment:
Don't bypass the interface of Scope here and access globals directly. Scope should either expose a ```load_global```, or ```Scope::load_name``` should accept a ```bytecode::Scope``` as a parameter. (In which case ```Scope::load_cell```should merge into ```Scope::load_name```.)

Reply:
Okay, I'll go for `load_global` in that case. Should `globals` be private?

Reply:
It's needed by the ```globals``` builtin too. We could provide a getter method for that if we want.